File: src/java/tusan/nodes/base/Return.java
package net.tutla.tums.tusan.nodes.base;

import net.tutla.tums.tusan.Node;
import net.tutla.tums.tusan.lexer.Token;
import net.tutla.tums.tusan.nodes.expression.Expression;

public class Return extends Node {
    public Return(Token token){
        super(token);
    }

    public Return create(){
        this.interpreter.returned = new Expression(token).create().value;
        this.interpreter.end = true;
        return this;
    }
}

File: src/python/src/tusan/nodes/base/Return.py
from tusan.Node import *
from tusan.nodes.expression import *


class ReturnNode(Node):
    def __init__(self, token:Token):
        self.interpreter = token.interpreter
        self.token = token

    async def create(self):
        self.interpreter.return_value = (await ExpressionNode(self.interpreter.next_token()).create()).value
        self.interpreter.end_found = True
        return self

        
